lighthouse-core/audits/is-on-https.js

```diff
@@ -72,6 +72,11 @@ class HTTPS extends Audit {
       extendedInfo: {
         formatter: Formatter.SUPPORTED_FORMATS.URL_LIST,
         value: insecureRecords
+      },
+      details: {
```

Comment:
I'm in favor of keeping extendedInfo and details as separate things. (The names could change though).

I really like that extendedInfo can hold all the metadata so downstream tools can consume and analyze the extra information we have on the audit. 

For example, pwmetrics, lighthouse-plots are consuming data that's in extendedInfo. And `perf-battle` is, as well: https://github.com/shakyShane/perf-battle/blob/31943652eb08c7bb1047805032b99d0aff82ad11/src/output-size.ts#L17

I think its wise for this AuditResult object to cleanly separate the data we need to render the report (in a reliable structure) vs the extra data we can offer (with no structure guarantees).


-------

> Not sure we need recursive rendering of details just yet 

how about datagrid where the first column is a URL? URL will likely need its own `render` as it'll get forked for devtools revealing.